cmd/snapshot-controller/main.go

```diff
@@ -64,18 +64,17 @@ var (
 )
 
 // Checks that the VolumeSnapshot v1 CRDs exist.
-func ensureCustomResourceDefinitionsExist(kubeClient *kubernetes.Clientset, client *clientset.Clientset) error {
+func ensureCustomResourceDefinitionsExist(client *clientset.Clientset) error {
 	condition := func() (bool, error) {
 		var err error
-		_, err = kubeClient.CoreV1().Namespaces().Get(context.TODO(), "kube-system", metav1.GetOptions{})
-		if err == nil {
-			// only execute list VolumeSnapshots if the kube-system namespace exists
-			_, err = client.SnapshotV1().VolumeSnapshots("kube-system").List(context.TODO(), metav1.ListOptions{})
-			if err != nil {
-				klog.Errorf("Failed to list v1 volumesnapshots with error=%+v", err)
-				return false, nil
-			}
+
+		// scoping to an empty namespace makes `List` work across all namespaces
+		_, err = client.SnapshotV1().VolumeSnapshots("").List(context.TODO(), metav1.ListOptions{})
```

Comment:
If there is no RBAC rule for "list" VolumeSnapshots, this will also fail.  Isn't this a similar problem as "get"?